### ISSUE TITLE
Request to Update Anchor Behavior and Guidance

### ISSUE THREAD

**Issue by derekhchen** [grommet/hpe-design-system#1847]
### Is this issue related to the design system core library or design system website?

Both -- I'm requesting DS to review the guidance on the use of anchors.

### Expected Behavior

It'd be great if anchors can be used to trigger actions instead of just for hyperlinks. I'd argue that the average user is not aware of the difference. The restriction that we have seems to be purely for design and dev to know when to use it vs a button. 

### Actual Behavior

Anchor links can only be used for hyperlinks.

### URL, screen shot, or CodeSandbox exhibiting the issue

![image](https://user-images.githubusercontent.com/38574571/132886980-ae461328-1d9c-4350-95e8-048e0d9c9a92.png)
In this example you can imagine that the user doesn't know or wouldn't care whether the link redirects them to another page or if it opens a modal with a form to submit the issue. The benefit of allowing anchor links to be used this way is that it's a lot more contextual and user-friendly, if used properly.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Design System version:
- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by samuelgoff** [comment]
@derekhchen, the existing guidance is based on a convention that's been in place since the beginning of the Web:
- underlined text -- aka hyperlinks -- cause a change in navigation, like flipping the pages in a book or jumping to a different page in a completely different book
- buttons trigger an action/function, like pushing the button on your stove to turn in on or off

Sometimes a button can do both: cause navigation + trigger an action/function. In the example you give, logging in is an action, but it can also result in a change in navigation. No biggie, IMO. I don't think any user is going to be given whiplash as a result of this combined behavior.

However, the opposite is not true: hyperlinks should not trigger an action + navigation. If they do, then this is an indication of poor implementation.

There is behavior that web browsers give you for free associated with anchors -- e.g. hover states & route changes based on `href`) that will actually **_break_** the implementation if you use anchors to trigger a behavior (i.e. javascript) without also interrupting this default behavior. This can lead to all kinds of odd bugs, including clicking an anchor-as-button that is supposed to trigger a UI change, but instead opens into a new window or other replaces the existing web app with another (incorrect) page. 🤮

Then there are also accessibility considerations for use of the wrong semantic element for the job. Assistive technologies behave **much** better when you use the right semantic elements in the right places, obviating the need to "fix" what isn't broken with ARIA. What designers see as a styling choice, UI implementers and Assistive technologies see as functional considerations. If you do nothing other than simply use the right semantic element for the job everywhere in an app or web site, you automatically comply with vastly more requirements to achieve WCAG 2.1/2.2 at the AA and even AAA standard, freeing you to focus on addressing a much smaller list that remains to achieve, maintain, and increase accessibility compliance.

Users have varying degrees of awareness of how these conventions work. For example, for some users, every single click is a double-click. It doesn't matter what they are clicking; they always double-click, even though it slows them down and sometimes forces them to double-click multiple times, since the thing they they're double-clicking is actually a toggle or hyperlink, and successive clicks interrupt the previous ones. Do we change all our apps and UX designs to make it so that every click required is a double-click? No, most emphatically not. Whether all users are aware of the convention or not, these Web standards exist for a reason.

**Comment by amandadupell** [comment]
Took a look at this with Bill today to get his opinion on the accessibility aspect to this issue. To @samuelgoff's comment above, it is best practice and the current industry standard to "use links when the user-action doesn’t affect the website at all. In this, the users are just readers or spectators of the site. For example, to navigate to the next page or an external source after viewing the content of the page". Similarly, "use buttons when the user- action causes a change in either back-end or the front-end of the website. For example, submitting a form, opening a pop-up or a modal or a pannel on the same page". Here is the link to that information: https://www.digitala11y.com/links-vs-buttons-a-perennial-problem/#:~:text=Use%20buttons%20when%20the%20user,or%20spectators%20of%20the%20site

Additionally, Bill and I spoke on the purpose of the Design System and good accessibility guidance. In general, it is always better to establish a system and remain consistent across products. This way, a user can familiarize themselves with how elements of an application work and anticipate the same behavior when they visit other pages/products. The most important thing regarding this issue is consistent guidance across HPE. Since the DS serves as this source of truth for developers and designers within the company, changing the guidance would break this consistency. I think it is best to establish a pattern and rules and stick with them.

**Comment by ericsoderberghp** [comment]
One thing to clarify, grommet's Anchor and Button components refer to the visual representation of the element. Either of them can be rendered to the DOM as an `<a />` or a `<button />` as needed to achieve the above behavior for accessibility.

**Comment by samuelgoff** [comment]
Hi @amandadupell, it sounds like we are using different words to say the same thing about the distinction between links (anchors) vs buttons.

I agree with you, Bill, & others in the need for consistency -- not just with the Design System itself, but also consistency with expectations & standards that have grown up since the beginning of the Web (actually, earlier). Users have been **trained for decades** to recognize underlined text as navigable text, which you can right-click to open in a new tab/window, command-click (on Mac) or shift-click on Windows to pop open into a new tab/window, etc. All of these expectations and behaviors get hijacked when you use that styling for a button, which no longer operates the same way.

In particular, I often command-click to pop open multiple links from a list view or dashboard (as do many power users), so I can use the closing of the tab as a way of keeping track of which options I've explored and which remain to be examined now or in the future. For example, when it comes to the GLC dashboard, it is not possible to do this. Try it yourself: right-click a widget on the GLC dashboard, and note it doesn't allow you to "Open Link in New Tab" or  "Open Link in New Window". This is particularly frustrating for some of us users -- because the semantic Anchor tag was not used for the cards, even though the result was a change in navigation.

@ericsoderberghp, thanks for pointing this out. By allowing Design System consumers to create something that **_looks like an anchor but behaves like a button_** -- or **_looks like a button but behaves like an anchor_** -- we are being inconsistent with standards and expectations of the Web (and Web apps).

By allowing Design System consumers to have "dealer's choice" on whether something is an `<a />` or `<button />`, we **guarantee** inconsistencies in how they are presented to end users. I would think this is exactly the type of thing we want to avoid -- by design, not by happy accident.

**Comment by ericsoderberghp** [comment]
I agree with the desire to command-click to open a new tab for the GLC Dashboard cards. And, it is for that reason where the card should be rendered as an `<a />` in the DOM. But, are you saying that the text in the dashboard card should be underlined? I think it is fine for the dashboard cards to be rendered visually like they are but to use a `<a />` in the DOM for the behavioral reasons you mention.

**Comment by samuelgoff** [comment]
Hey @ericsoderberghp, good point. I can understand this might have left room for confusion, since I was talking about both styling and behavior in the same response. In the case of the widget/card, I was talking about the **_behavior_** users are expecting from navigable links (command-click, etc.). In this case, the entire card, which IMO looks and acts like a [`<figure />`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure) or clickable image, which is yet another convention users are completely trained to recognize. Just because a figure / card / widget is navigable by clicking **_and_** contains text, doesn't mean every bit of text nested inside should be underlined. In fact, having large blocks of text underlined undermines their readability. The widgets/cards have a hover state (& cursor change), which suggests they are "alive" -- this conveys to the user that clicking on them will produce a result.

**Comment by samuelgoff** [comment]
@ericsoderberghp So to clarify, I think we are in complete agreement about styling (no underline) but using the `<a />` to wrap them when they render into the DOM.

**Comment by MikeKingdom** [comment]
If a card takes you to some other page for something (which is very typical) I definitely want the ability to right-click/open in a new tab. I do it all the time.
